service/src/main/kotlin/xyz/lebkuchenfm/external/discord/DiscordClient.kt

```diff
@@ -53,13 +59,41 @@ class DiscordClient(config: ApplicationConfig, private val commandExecutorServic
             .map { it.message }
             .filter { it.channelId.value.toString() == channelId }
             .filter { it.content.startsWith(commandPrompt) }
+            .filter { it.author != null }
             .filter { it.author?.isBot == false }
             .onEach {
-                // TODO: Get user by Discord ID.
-                val context = ExecutionContext(UserSession(name = "FAKE USER TODO"))
-                val result = commandExecutorService.executeFromText(it.content, context)
-                it.reply {
-                    content = result.message.markdown
+                val author = requireNotNull(it.author)
+                val user = userService.getByDiscordId(author.id.toString())
+
+                // TODO: Add support for login command.
+                val isLoginCommand = false
+
+                when {
+                    isLoginCommand -> {
+                        val context = ExecutionContext(UserSession(name = "unknown"))
```

Comment:
Shouldn't session in context be optional for such a case?

Reply:
But this is only the case for login command and it does not use session data. Making it optional will make us do useless not null check all over the code because of this.
I'm personally more interested in changing the Discord auth flow entirely (removing login command), and making this field optional is not worth it in my opinion.

Reply:
k, but having "unknown" passed as a name on production smells 💩

Reply:
also write down to the doc that part about rewriting discord auth flow ;)

Reply:
We agreed not to do it that way. We will worry about login command when we actually implement it.